### PR TITLE
Add stock _inventory_import (migrated from odoomrp-wip)

### PR DESCRIPTION
**Pull request by tafaRU**
This PR contains the module 'stock_inventory_import' migrated from https://github.com/odoomrp/odoomrp-wip according to https://github.com/odoomrp/odoomrp-wip/wiki/Migration-to-OCA

### CODE REVIEW

File: stock_inventory_import/README.rst [hunk 1]
```diff
@@ -0,0 +1,48 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
+
+Stock Inventory Import from CSV file
+====================================
+
+Wizard to import Inventory from a CSV file
+The file must have at least 2 columns with "code" and "quantity" Head Keys.
```
Comment (pedrobaeza) [inline stock_inventory_import/README.rst:14]: This can be in the "Configuration" section.

File: stock_inventory_import/models/__init__.py [hunk 2]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (pedrobaeza) [inline stock_inventory_import/models/__init__.py]: Put encoding

File: stock_inventory_import/models/stock_inventory.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
+#
```
Comment (pedrobaeza) [inline stock_inventory_import/models/stock_inventory.py]: Put encoding

File: stock_inventory_import/security/.directory [hunk 4]
```diff
@@ -0,0 +1,2 @@
+[Dolphin]
```
Comment (pedrobaeza) [inline stock_inventory_import/security/.directory]: Remove this file

File: stock_inventory_import/security/ir.model.access.csv [hunk 5]
```diff
@@ -0,0 +1,2 @@
+id,name,model_id:id,group_id:id,perm_read,perm_write,perm_create,perm_unlink
+access_stock_inventory_import_line,access_stock_inventory_import_line,model_stock_inventory_import_line,stock.group_stock_manager,1,1,1,1
```
Comment (pedrobaeza) [inline stock_inventory_import/security/ir.model.access.csv]: Put also read access to warehouse user, as inventory has.

File: stock_inventory_import/wizard/__init__.py [hunk 6]
```diff
@@ -0,0 +1,2 @@
+
```
Comment (pedrobaeza) [inline stock_inventory_import/wizard/__init__.py]: Put encoding

File: stock_inventory_import/wizard/import_inventory.py [hunk 7]
```diff
@@ -0,0 +1,104 @@
+
```
Comment (pedrobaeza) [inline stock_inventory_import/wizard/import_inventory.py]: Put encoding

File: stock_inventory_import/wizard/import_inventory.py [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+class ImportInventory(models.TransientModel):
+    _name = 'import.inventory'
+    _description = 'Import inventory'
+
+    def _get_default_location(self):
+        ctx = self._context
+        if 'active_id' in ctx:
+            inventory_obj = self.env['stock.inventory']
+            inventory = inventory_obj.browse(ctx['active_id'])
+        return inventory.location_id
```
Comment (pedrobaeza) [inline stock_inventory_import/wizard/import_inventory.py]: Indent this line one level more and add another one `return False` as fallback.

File: stock_inventory_import/wizard/import_inventory_view.xml [hunk 9]
```diff
@@ -0,0 +1,42 @@
...
+    <record id="action_import_inventory" model="ir.actions.act_window">
+        <field name="name">Import Inventory</field>
+        <field name="res_model">import.inventory</field>
+        <field name="view_type">form</field>
+        <field name="view_mode">tree,form</field>
+       <field name="view_id" ref="view_import_inventory"/>
+       <field name="target">new</field>
+    </record>
+
+<!--     <menuitem  id="menu_import_inventory" name="Import Inventory" action="action_import_inventory" parent="stock.menu_stock_inventory_control"/> -->
```
Comment (pedrobaeza) [inline stock_inventory_import/wizard/import_inventory_view.xml]: Remove this commented section

**Comment by tafaRU** [comment]
@pedrobaeza I think all your remarks have been addressed, could you please check it?
Thank you in advance.

**Comment by pedrobaeza** [comment]
:+1:

File: stock_inventory_import/README.rst [hunk 10]
```diff
@@ -0,0 +1,52 @@
...
+You can also add a fourth column with Key "lot" to add a product lot.
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/{stock-logistics-workflow}/issues>`_.
+In case of trouble, please check there if your issue has already been reported.
+If you spotted it first, help us smashing it by providing a detailed and welcomed feedback
+`here <https://github.com/OCA/{stock-logistics-workflow}/issues/new?body=module:%20{stock_inventory_import}%0Aversion:%20{version}%0A%0A**Steps%20to%20reproduce**%0A-%20...%0A%0A**Current%20behavior**%0A%0A**Expected%20behavior**>`_.
```
Comment (eLBati) [inline stock_inventory_import/README.rst]: please remove `{` `}`

Comment (tafaRU) [inline stock_inventory_import/README.rst]: @eLBati, done.

**Comment by eLBati** [comment]
:+1: thanks

**Comment by pedrobaeza** [comment]
Can you add tests for exercising this module?

**Comment by gurneyalex** [comment]
@tafaRU could you rebase the PR please?

**Comment by tafaRU** [comment]
> @tafaRU could you rebase the PR please?

@gurneyalex, done

File: stock_inventory_import/__openerp__.py [hunk 11]
```diff
@@ -0,0 +1,25 @@
+# -*- coding: utf-8 -*-
+# © 2015 AvanzOSC
+# © 2015-2016 Pedro M. Baeza <pedro.baeza@tecnativa.com>
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+
+{
+    "name": "Stock Inventory Import from CSV file",
+    "version": "1.0",
```
Comment (eLBati) [inline stock_inventory_import/__openerp__.py]: Please use 8.0.1.0.0

**Comment by eLBati** [comment]
@tafaRU could you rebase to see if tests errors related to `stock_available_mrp` are fixed?

**Comment by pedrobaeza** [comment]
There's a new commit solving a little issue in odoomrp-wip. Please cherry-pick.

File: stock_inventory_import/__openerp__.py (hunk 11 above)
Comment (tafaRU) [reply stock_inventory_import/__openerp__.py]: @eLBati, done at https://github.com/OCA/stock-logistics-warehouse/pull/56/commits/b2ef44224a3c1b8c52b90d9fa0fdf24a87d6dc27

**Comment by tafaRU** [comment]
@pedrobaeza, cherry-pick done, please see https://github.com/OCA/stock-logistics-warehouse/pull/56/commits/f210de422dab47633d34949707166414948b000a

File: stock_inventory_import/README.rst [hunk 12]
```diff
@@ -0,0 +1,52 @@
...
+The file must have at least 2 columns with "code" and "quantity" Head Keys.
+You can also add a third column with Key "location" to add product location
+(if not defined, default inventory location will be used)
+You can also add a fourth column with Key "lot" to add a product lot.
+
+
+Bug Tracker
+===========
+
+Bugs are tracked on `GitHub Issues <https://github.com/OCA/stock-logistics-warehouse/issues>`_.
```
Comment (andhit-r) [inline stock_inventory_import/README.rst]: Use new format, see https://github.com/OCA/maintainer-tools/blame/master/template/module/README.rst#L52

File: stock_inventory_import/README.rst [hunk 13]
```diff
@@ -0,0 +1,52 @@
...
+   :alt: Odoo Community Association
+   :target: https://odoo-community.org
+
+This module is maintained by the OCA.
+
+OCA, or the Odoo Community Association, is a nonprofit organization whose
+mission is to support the collaborative development of Odoo features and
+promote its widespread use.
+
+To contribute to this module, please visit http://odoo-community.org.
```
Comment (andhit-r) [inline stock_inventory_import/README.rst]: https

File: stock_inventory_import/__init__.py [hunk 14]
```diff
@@ -0,0 +1,20 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (andhit-r) [inline stock_inventory_import/__init__.py]: Use simple header

File: stock_inventory_import/__openerp__.py [hunk 15]
```diff
@@ -0,0 +1,42 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (andhit-r) [inline stock_inventory_import/__openerp__.py]: Use simple header

File: stock_inventory_import/__openerp__.py [hunk 16]
```diff
@@ -0,0 +1,42 @@
...
+#
+##############################################################################
+{
+    "name": "Stock Inventory Import from CSV file",
+    "version": "8.0.1.0.0",
+    "category": "Generic Modules",
+    "author": "OdooMRP team, "
+              "AvanzOSC, "
+              "Serv. Tecnol. Avanzados - Pedro M. Baeza",
+    "contributors": [
```
Comment (andhit-r) [inline stock_inventory_import/__openerp__.py]: no need, already on README

File: stock_inventory_import/models/__init__.py [hunk 17]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (andhit-r) [inline stock_inventory_import/models/__init__.py]: Use simple header

File: stock_inventory_import/models/stock_inventory.py [hunk 18]
```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (andhit-r) [inline stock_inventory_import/models/stock_inventory.py]: Use simple header

File: stock_inventory_import/wizard/__init__.py [hunk 19]
```diff
@@ -0,0 +1,19 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (andhit-r) [inline stock_inventory_import/wizard/__init__.py]: Use simple header

File: stock_inventory_import/wizard/import_inventory.py [hunk 20]
```diff
@@ -0,0 +1,122 @@
+# -*- coding: utf-8 -*-
+##############################################################################
```
Comment (andhit-r) [inline stock_inventory_import/wizard/import_inventory.py]: Use simple header

File: stock_inventory_import/wizard/import_inventory.py [hunk 21]
```diff
@@ -0,0 +1,122 @@
...
+        return False
+
+    data = fields.Binary('File', required=True)
+    name = fields.Char('Filename')
+    delimeter = fields.Char('Delimeter', default=',',
+                            help='Default delimeter is ","')
+    location = fields.Many2one('stock.location', 'Default Location',
+                               default=_get_default_location, required=True)
+
+    @api.one
```
Comment (andhit-r) [inline stock_inventory_import/wizard/import_inventory.py]: Use @api.multi with self.ensure_one()

File: stock_inventory_import/wizard/import_inventory.py [hunk 22]
```diff
@@ -0,0 +1,122 @@
...
+#    but WITHOUT ANY WARRANTY; without even the implied warranty of
+#    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+#    GNU Affero General Public License for more details.
+#
+#    You should have received a copy of the GNU Affero General Public License
+#    along with this program.  If not, see <http://www.gnu.org/licenses/>.
+#
+##############################################################################
+
+from openerp import fields, models, exceptions, api, _
```
Comment (andhit-r) [inline stock_inventory_import/wizard/import_inventory.py]: Use:

```
from openerp.exceptions import Warning as UserError 
```

as stated on https://github.com/OCA/maintainer-tools/blob/master/CONTRIBUTING.md#idioms

**Review by andhit-r — CHANGES_REQUESTED** [review]
Functionality: Ok (tested on runbot)

Code:
1. Travis is still failed
2. Misused @api.one insted @api.multi
3. Minor details

**Comment by tafaRU** [comment]
@pedrobaeza, a couple of questions:
- which author can I use into the header?
- does OCA need to be added as author in the manifest file?

Thank you in avance.

**Comment by pedrobaeza** [comment]
You can add `Tecnativa` and `AvanzOSC` as original authors.

You should add OCA also as co-author.

**Comment by tafaRU** [comment]
@andhit-r, I addressed your remarks, please have a look at it.

**Review by andhit-r — APPROVED** [review]
:+1:

**Comment by tafaRU** [comment]
PR is ready to be merged! Could someone of @OCA/logistics-maintainers please do it?
Thank you in advance.

**Review by pedrobaeza — APPROVED** [review]
Merging and amending some details after (author, module category and license link)

**Review by pedrobaeza — CHANGES_REQUESTED** [review]
Hey, wait a moment, Travis status is red. Please check that. You can also amend the rest of the commented things

File: stock_inventory_import/README.rst [hunk 23]
```diff
@@ -0,0 +1,52 @@
+.. image:: https://img.shields.io/badge/licence-AGPL--3-blue.svg
+    :alt: License: AGPL-3
```
Comment (pedrobaeza) [inline stock_inventory_import/README.rst]: Add license link

File: stock_inventory_import/__openerp__.py [hunk 24]
```diff
@@ -0,0 +1,24 @@
+# -*- coding: utf-8 -*-
+# Copyright 2016 Avanzosc (<http://www.avanzosc.es>)
+# Copyright 2016 Tecnativa (<http://www.tecnativa.com>)
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl).
+{
+    "name": "Stock Inventory Import from CSV file",
+    "version": "8.0.1.0.0",
+    "category": "Generic Modules",
```
Comment (pedrobaeza) [inline stock_inventory_import/__openerp__.py]: Change this to "Warehouse Management"

File: stock_inventory_import/__openerp__.py [hunk 25]
```diff
@@ -0,0 +1,24 @@
...
+    "author": "OdooMRP team, "
+              "AvanzOSC, "
+              "Serv. Tecnol. Avanzados - Pedro M. Baeza, "
```
Comment (pedrobaeza) [inline stock_inventory_import/__openerp__.py]: Change this to `Tecnativa`

**Comment by tafaRU** [comment]
@pedrobaeza, I fixed the things you commented.
Travis is still red but I don't think it's related to this module especially because it doesn't contain any tests.

**Comment by pedrobaeza** [comment]
Please rebase on latest 8.0 branch to see

**Comment by tafaRU** [comment]
@pedrobaeza, rebased but travis is still red :expressionless:

**Comment by tafaRU** [comment]
Travis will be green once https://github.com/OCA/stock-logistics-warehouse/pull/140 will be merged.

**Comment by JordiBForgeFlow** [comment]
@tafaRU I have now merged https://github.com/OCA/stock-logistics-warehouse/pull/140 and restarted Travis for this PR.

**Comment by tafaRU** [comment]
Thanks @jbeficent, I've just rebased, let's see!

**Comment by eLBati** [comment]
still red

**Comment by Cedric-Pigeon** [comment]
@tafaRU Hello, do you plan to achieve this PR ? Is it still usefull ? thanks

**Comment by pedrobaeza** [comment]
No answer, so I close it. Feel free to reopen if needed.

**Pull request closed without merging**